File: app/src/main/java/com/konektedi/vs/Home/Categories/CategoriesRepository.java
package com.konektedi.vs.Home.Categories;

import android.arch.lifecycle.MutableLiveData;

import com.konektedi.vs.Utilities.ApiUtilities;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class CategoriesRepository {

    public MutableLiveData<List<CategoriesModel>> getCategories(String election_id) {

        final MutableLiveData<List<CategoriesModel>> listMutableLiveData = new MutableLiveData<>();

        Call<List<CategoriesModel>> call = ApiUtilities.getCategories().getCategories(election_id);

        call.enqueue(new Callback<List<CategoriesModel>>() {
            @Override
            public void onResponse(Call<List<CategoriesModel>> call, Response<List<CategoriesModel>> response) {
                if (response.isSuccessful()) {
                    List<CategoriesModel> categoriesModelList = response.body();

                    listMutableLiveData.setValue(categoriesModelList);

                }

            }

            @Override
            public void onFailure(Call<List<CategoriesModel>> call, Throwable t) {

            }

        });

        return listMutableLiveData;

    }
}

File: app/src/main/java/com/konektedi/vs/home/categories/CategoriesRepository.kt
package com.konektedi.vs.home.categories

import android.arch.lifecycle.MutableLiveData
import com.konektedi.vs.utilities.api.ApiN
import com.konektedi.vs.utilities.common.NetworkState
import com.konektedi.vs.utilities.models.Category
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class CategoriesRepository {

    private val apiClient = ApiN.create()
    val mNetworkState = MutableLiveData<NetworkState>()

    fun getCategories(electionId: Int): MutableLiveData<List<Category>> {
        mNetworkState.value = NetworkState.LOADING
        val categoriesList = MutableLiveData<List<Category>>()

        apiClient.getCategories(electionId).enqueue(
                object : Callback<List<Category>> {
                    override fun onFailure(call: Call<List<Category>>, t: Throwable) {
                        mNetworkState.value = NetworkState.FAILED
                    }

                    override fun onResponse(call: Call<List<Category>>, response: Response<List<Category>>) {
                        when {
                            response.isSuccessful -> {
                                mNetworkState.value = NetworkState.LOADED
                                categoriesList.value = response.body()
                            }
                            response.code() == 404 -> mNetworkState.value = NetworkState.END
                            else -> {
                                mNetworkState.postValue(NetworkState.error("error code: ${response.code()}"))
                            }
                        }
                    }
                })
        return categoriesList
    }


}